lib/java/src/main/java/org/apache/thrift/TServiceClient.java

```diff
@@ -93,4 +95,12 @@ protected void receiveBase(TBase<?, ?> result, String methodName) throws TExcept
     result.read(iprot_);
     iprot_.readMessageEnd();
   }
+
+  protected void after() {
+    TTransport transport = this.iprot_.getTransport();
+    if (transport instanceof TFramedTransport) {
+      TFramedTransport framedTransport = (TFramedTransport) transport;
+      framedTransport.clear();
```

Comment:
This can be simplified

```suggestion
      ((TFramedTransport) transport).clear();
```